src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler_Return.cs

```diff
@@ -50,12 +51,233 @@ public bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext co
+
+        var token = parsedDocument.Root.FindToken(position);
+        var preferredIndentationToken = token;
+        switch (token.Kind())
+        {
+            case SyntaxKind.SingleLineRawStringLiteralToken:
+                break;
+
+            case SyntaxKind.InterpolatedStringTextToken:
+            case SyntaxKind.OpenBraceToken:
+                if (token.Parent?.Parent is InterpolatedStringExpressionSyntax interpolated &&
+                    interpolated.StringStartToken.Kind() is SyntaxKind.InterpolatedSingleLineRawStringStartToken)
+                {
+                    if (token.Kind() is SyntaxKind.OpenBraceToken)
+                    {
+                        // If we are not at the start of the interpolation braces, we do not intend to handle converting the raw string
+                        // into a new one
+                        if (position != token.SpanStart)
+                        {
+                            return false;
+                        }
+
+                        // We prefer the indentation options of the string start delimiter because the indentation of the interpolation
+                        // is empty and thus we cannot properly indent the lines that we insert
+                        preferredIndentationToken = interpolated.StringStartToken;
+                    }
+
+                    break;
+                }
+
+                return false;
+
+            default:
+                return false;
+        }
+
+        return MakeEdit(textView, subjectBuffer, position, document, parsedDocument, token, preferredIndentationToken, isEmpty: false, cancellationToken);
+    }
+
+    private bool MakeEdit(
+        ITextView textView, ITextBuffer subjectBuffer, int position, Document document,
+        ParsedDocument parsedDocument, SyntaxToken token, SyntaxToken preferredIndentationToken,
+        bool isEmpty, CancellationToken cancellationToken)
+    {
+        var indentationOptions = subjectBuffer.GetIndentationOptions(_editorOptionsService, document.Project.GetFallbackAnalyzerOptions(), document.Project.Services, explicitFormat: false);
+        var indentation = preferredIndentationToken.GetPreferredIndentation(parsedDocument, indentationOptions, cancellationToken);
+
+        var newLine = indentationOptions.FormattingOptions.NewLine;
+        using var transaction = CaretPreservingEditTransaction.TryCreate(
+            CSharpEditorResources.Split_raw_string, textView, _undoHistoryRegistry, _editorOperationsFactoryService);
+        var edit = subjectBuffer.CreateEdit();
+
+        var openingEnd = GetEndPositionOfOpeningDelimiter(token, isEmpty);
+        var insertedLines = 1;
+        if (isEmpty)
+        {
+            edit.Insert(position, newLine + newLine + indentation);
+        }
+        else
+        {
+            var newLineAndIndentation = newLine + indentation;
+            // Add a newline at the position of the end literal
```

Comment:
this is just stating what hte code is doing.  please state why it is doing that.